assets/dev/js/frontend/handlers/image-carousel.js

```diff
@@ -2,63 +2,132 @@ class ImageCarouselHandler extends elementorModules.frontend.handlers.Base {
 	getDefaultSettings() {
 		return {
 			selectors: {
-				carousel: '.elementor-image-carousel',
+				carousel: '.elementor-image-carousel-wrapper',
+				slideContent: '.swiper-slide',
 			},
 		};
 	}
 
 	getDefaultElements() {
 		const selectors = this.getSettings( 'selectors' );
 
-		return {
+		const elements = {
 			$carousel: this.$element.find( selectors.carousel ),
 		};
+
+		elements.$swiperSlides = elements.$carousel.find( selectors.slideContent );
+
+		return elements;
 	}
 
-	onInit( ...args ) {
-		super.onInit( ...args );
+	getSlidesCount() {
+		return this.elements.$swiperSlides.length;
+	}
 
+	getSwiperSettings() {
 		const elementSettings = this.getElementSettings(),
 			slidesToShow = +elementSettings.slides_to_show || 3,
 			isSingleSlide = 1 === slidesToShow,
 			defaultLGDevicesSlidesCount = isSingleSlide ? 1 : 2,
-			breakpoints = elementorFrontend.config.breakpoints;
-
-		const slickOptions = {
-			slidesToShow: slidesToShow,
-			autoplay: 'yes' === elementSettings.autoplay,
-			autoplaySpeed: elementSettings.autoplay_speed,
-			infinite: 'yes' === elementSettings.infinite,
-			pauseOnHover: 'yes' === elementSettings.pause_on_hover,
+			elementorBreakpoints = elementorFrontend.config.breakpoints;
+
+		const swiperOptions = {
+			slidesPerView: slidesToShow,
+			slidesPerGroup: +elementSettings.slides_to_scroll || 1,
+			loop: 'yes' === elementSettings.infinite,
 			speed: elementSettings.speed,
-			arrows: -1 !== [ 'arrows', 'both' ].indexOf( elementSettings.navigation ),
-			dots: -1 !== [ 'dots', 'both' ].indexOf( elementSettings.navigation ),
-			rtl: 'rtl' === elementSettings.direction,
-			responsive: [
-				{
-					breakpoint: breakpoints.lg,
-					settings: {
-						slidesToShow: +elementSettings.slides_to_show_tablet || defaultLGDevicesSlidesCount,
-						slidesToScroll: +elementSettings.slides_to_scroll_tablet || defaultLGDevicesSlidesCount,
-					},
-				},
-				{
-					breakpoint: breakpoints.md,
-					settings: {
-						slidesToShow: +elementSettings.slides_to_show_mobile || 1,
-						slidesToScroll: +elementSettings.slides_to_scroll_mobile || 1,
-					},
-				},
-			],
 		};
 
-		if ( isSingleSlide ) {
```

Comment:
Did you applied this functionality in the new implementation?

Reply:
No. The Image Carousel widget does not have an option for controlling the scrolling effect (it is always 'slide'). So the contents in this `if` statement were never executed anyway.
The `slidesToScroll` are handled.

Reply:
Return it back